evalml/data_checks/datetime_format_data_check.py

```diff
@@ -118,21 +179,66 @@ def validate(self, X, y):
             )
             return messages
 
-        if not inferred_freq:
-            col_name = (
-                self.datetime_column
-                if self.datetime_column != "index"
-                else "either index"
-            )
+        col_name = (
+            self.datetime_column if self.datetime_column != "index" else "either index"
+        )
+
+        is_increasing = pd.DatetimeIndex(datetime_values).is_monotonic_increasing
+
+        nan_columns = datetime_values.isna().any()
+        if nan_columns:
             messages.append(
                 DataCheckError(
-                    message=f"No frequency could be detected in {col_name}, possibly due to uneven intervals.",
+                    message=f"Input datetime column ({col_name}) contains NaN values. Please impute NaN values or drop these rows.",
                     data_check_name=self.name,
-                    message_code=DataCheckMessageCode.DATETIME_HAS_UNEVEN_INTERVALS,
+                    message_code=DataCheckMessageCode.DATETIME_HAS_NAN,
                 ).to_dict()
             )
+            # NaN values are a more significant error than missing values and will break the data check logic, so we return here
+            return messages
+
+        if not inferred_freq:
 
-        if not (pd.DatetimeIndex(datetime_values).is_monotonic_increasing):
+            # Check for only one row per datetime
+            duplicate_dates = pd.Series(datetime_values).diff(1) == pd.Timedelta(0)
+            if duplicate_dates.any():
+                messages.append(
+                    DataCheckError(
+                        message=f"{col_name} has more than one row with the same datetime value.",
```

Comment:
I think the only thing these messages need is maybe a bit more clarity like "Column '{col_name}' has...".  I think, without the single quotes, it's not entirely clear that it's a column name and not some var.